### PR TITLE
Port Microsoft.Extensions.* from pkgproj to dotnet pack

### PR DESCRIPTION
**Pull request by Anipik**
Design Changes
- no more pkgproj
- separate symbol nuspec files
- minor change in metadata like no owners (Metadata not supported by the nuget pack)
- nuspec name contains package version
- no more pkgpath files (not sure why we were producing it in the first place)
- xml files comes from the build and not from intellisense package (cc @safern is that fine ?)
- no more package reports
- removed implicit framework assembly references from the package
- symbol package path will be same as package path 
- exclude="Build,Analyzers"  for all dependencies

Validation changes
- Closure verification in the all config
- Package validation during build packages is no longer done as it was not doing any substantial work for extensions packages.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -222,10 +222,16 @@
...
+    <PackageProjectUrl>$(ProjectUrl)</PackageProjectUrl>
+    <PackageIconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</PackageIconUrl>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
     <Copyright>$(CopyrightNetFoundation)</Copyright>
     <PackageThirdPartyNoticesFile>$(MSBuildThisFileDirectory)THIRD-PARTY-NOTICES.TXT</PackageThirdPartyNoticesFile>
+    <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
     <ReleaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</ReleaseNotes>
+    <PackageReleaseNotes>$(ReleaseNotes)</PackageReleaseNotes> 
+    <Title>$(MSBuildProjectName)</Title>
+    <MinClientVersion>2.12</MinClientVersion>
```
Comment (ViktorHofer) [inline Directory.Build.props:234]: I don't think we need that nuspec metadata anymore.

File: Directory.Build.props [hunk 2]
```diff
@@ -222,10 +222,16 @@
...
     <LicenseFile>$(MSBuildThisFileDirectory)LICENSE.TXT</LicenseFile>
+    <PackageProjectUrl>$(ProjectUrl)</PackageProjectUrl>
+    <PackageIconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</PackageIconUrl>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
     <Copyright>$(CopyrightNetFoundation)</Copyright>
     <PackageThirdPartyNoticesFile>$(MSBuildThisFileDirectory)THIRD-PARTY-NOTICES.TXT</PackageThirdPartyNoticesFile>
+    <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
     <ReleaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</ReleaseNotes>
+    <PackageReleaseNotes>$(ReleaseNotes)</PackageReleaseNotes> 
+    <Title>$(MSBuildProjectName)</Title>
```
Comment (ViktorHofer) [inline Directory.Build.props:233]: That property should be unnecessary as Title already defaults to PackageId which defaults to the project's name.

File: Directory.Build.props [hunk 3]
```diff
@@ -222,10 +222,16 @@
     <LicenseUrl>https://github.com/dotnet/$(GitHubRepositoryName)/blob/master/LICENSE.TXT</LicenseUrl>
     <RuntimeIdGraphDefinitionFile>$([MSBuild]::NormalizePath('$(LibrariesProjectRoot)', 'Microsoft.NETCore.Platforms', 'pkg', 'runtime.json'))</RuntimeIdGraphDefinitionFile>
     <LicenseFile>$(MSBuildThisFileDirectory)LICENSE.TXT</LicenseFile>
+    <PackageProjectUrl>$(ProjectUrl)</PackageProjectUrl>
+    <PackageIconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</PackageIconUrl>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
     <Copyright>$(CopyrightNetFoundation)</Copyright>
     <PackageThirdPartyNoticesFile>$(MSBuildThisFileDirectory)THIRD-PARTY-NOTICES.TXT</PackageThirdPartyNoticesFile>
+    <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
```
Comment (ViktorHofer) [inline Directory.Build.props:230]: That's already the default value. You can remove the property.

File: src/libraries/Directory.Build.targets [hunk 4]
```diff
@@ -231,6 +232,9 @@
 
   <PropertyGroup>
     <ExcludeFromPackage Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', '$(NetCoreAppCurrent)')) and '$(ExcludeCurrentNetCoreAppFromPackage)' == 'true'">true</ExcludeFromPackage>
+    <AllowedOutputExtensionsInPackageBuildOutputFolder>.dll;.exe;.winmd;.json;.pri;.xml;</AllowedOutputExtensionsInPackageBuildOutputFolder>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets:235]: Why is this necessary? I see that property already being defined with the same value in NuGet.targets: https://github.com/NuGet/NuGet.Client/blob/8eefb1e7d99c2992497b929c51e54b3c738b3e3b/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets#L48

File: src/libraries/Directory.Build.targets [hunk 5]
```diff
@@ -231,6 +232,9 @@
...
+    <IncludeSymbols>true</IncludeSymbols>
+    <PackageVersion Condition="'$(StablizePackageVersion)' != 'true'">$(PackageVersion)-$(VersionSuffix)</PackageVersion>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets:237]: That property is already defined here: https://github.com/dotnet/runtime/blob/1df2bc9018846da1b1ba4ba38cf04273388c7c27/eng/packaging.props#L13. We probably don't want yet another different meaning of it as we already have four of it in the repo?

**Review by ViktorHofer — COMMENTED** [review]
Can you please also inline the PackageDescription from the leaf's Directory.Build.props file into the csproj files that are now packable?

**Comment by ViktorHofer** [comment]
> no more pkgpath files (not sure why we were producing it in the first place)

What's a pkgpath file?

> minor change in metadata like no owners (Metadata not supported by the nuget pack)

Can you please share a diff of the nuspecs before and after?

File: src/libraries/Directory.Build.targets (hunk 4 above)
Comment (Anipik) [reply src/libraries/Directory.Build.targets:235]: This is necessary to avoid treating pdbs as build output and stop them from including in the packages.

File: src/libraries/Directory.Build.targets (hunk 5 above)
Comment (Anipik) [reply src/libraries/Directory.Build.targets:237]: This is transitory fix, First package version is consumed by the old infra and this one is consumed by the pack task.
We can merge them after we have ported all the projects to dotnet pack

File: Directory.Build.props [hunk 6]
```diff
@@ -222,10 +222,16 @@
     <LicenseUrl>https://github.com/dotnet/$(GitHubRepositoryName)/blob/master/LICENSE.TXT</LicenseUrl>
     <RuntimeIdGraphDefinitionFile>$([MSBuild]::NormalizePath('$(LibrariesProjectRoot)', 'Microsoft.NETCore.Platforms', 'pkg', 'runtime.json'))</RuntimeIdGraphDefinitionFile>
     <LicenseFile>$(MSBuildThisFileDirectory)LICENSE.TXT</LicenseFile>
+    <PackageProjectUrl>$(ProjectUrl)</PackageProjectUrl>
+    <PackageIconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</PackageIconUrl>
```
Comment (eerhardt) [inline Directory.Build.props:226]: I thought PackageIconUrl gave a warning now and instead you are supposed to use an embeded Icon.

[https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu5048](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu5048)

File: src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Microsoft.Extensions.Caching.Abstractions.csproj [hunk 7]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
     <EnableDefaultItems>true</EnableDefaultItems>
+    <IsPackable>true</IsPackable>
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Microsoft.Extensions.Caching.Abstractions.csproj:6]: Can we instead default this in a common props/targets file?

```xml
<IsPackable Condition="ProjectName.StartsWith('Microsoft.Extensions.') AND $(IsSourceProject)">true</IsPackable>
```

File: src/libraries/Directory.Build.targets (hunk 5 above)
Comment (safern) [reply src/libraries/Directory.Build.targets:237]: We should leverage the `PackageVersion` logic to arcade. I don't think we should be duplicating logic by checking `StabilizePackageVersion` and setting the `VersionSuffix` ourselves.

The way arcade does it is by appending `VersionSuffix` always if it is not empty, because it has a lot of calculations to decide whether we need a `VersionSuffix` or not:
https://github.com/dotnet/arcade/blob/99b9f805da070d301d76eac578a46dd75d9e5f7c/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets#L165

File: src/libraries/Directory.Build.targets (hunk 4 above)
Comment (safern) [reply src/libraries/Directory.Build.targets:235]: Wouldn't we need to use this property to produce a symbol package and that way the pdbs are not in the regular package? https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#includesymbols

Comment (Anipik) [reply src/libraries/Directory.Build.targets:235]: no, IncludeSymbols property enables building another package with symbols in it.
arcade changes the value of this property ```AllowedOutputExtensionsInPackageBuildOutputFolder```  from the default

Comment (safern) [reply src/libraries/Directory.Build.targets:235]: Right, but that is a fallback fo `IncludeSymbols` not being set to true: https://github.com/dotnet/arcade/blob/ca7af8d541882be495d710b0aaf28276110aa8fc/src/Microsoft.DotNet.Arcade.Sdk/tools/Workarounds.targets#L59

File: Directory.Build.props (hunk 2 above)
Comment (Anipik) [reply Directory.Build.props:233]: The title is not getting set without this.

File: Directory.Build.props (hunk 3 above)
Comment (Anipik) [reply Directory.Build.props:230]: the default value is true and set by projectDefault.props.

File: Directory.Build.props (hunk 2 above)
Comment (ViktorHofer) [reply Directory.Build.props:233]: You can just remove that property at all. If Title isn't provided, the PackageId which defaults to the project's name will be used. If a package wants to specify a title, they can do so in in the csproj directly.

**Comment by Anipik** [comment]
Sample diff, we will have a different nuspec file for symbols packages.


```diff

 <EF><BB><BF><?xml version="1.0" encoding="utf-8"?>
-<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
-  <metadata minClientVersion="2.12">
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
     <id>Microsoft.Extensions.Configuration.Binder</id>
     <version>6.0.0-dev</version>
     <title>Microsoft.Extensions.Configuration.Binder</title>
     <authors>Microsoft</authors>
-    <owners>microsoft,dotnetframework</owners>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <license type="expression">MIT</license>
     <licenseUrl>https://licenses.nuget.org/MIT</licenseUrl>
     <icon>Icon.png</icon>
     <projectUrl>https://github.com/dotnet/runtime</projectUrl>
-    <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
     <description>Functionality to bind an object to data in configuration providers for Microsoft.Extensions.Configuration.</description>
     <releaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</releaseNotes>
     <copyright><C2><A9> Microsoft Corporation. All rights reserved.</copyright>
-    <serviceable>true</serviceable>
     <repository type="git" url="git://github.com/dotnet/runtime" commit="0000000000000000000000000000000000000000" />
     <dependencies>
       <group targetFramework=".NETFramework4.6.1">
-        <dependency id="Microsoft.Extensions.Configuration.Abstractions" version="6.0.0-dev" />
+        <dependency id="Microsoft.Extensions.Configuration.Abstractions" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
       <group targetFramework=".NETStandard2.0">
-        <dependency id="Microsoft.Extensions.Configuration.Abstractions" version="6.0.0-dev" />
+        <dependency id="Microsoft.Extensions.Configuration.Abstractions" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
     </dependencies>
-    <frameworkAssemblies>
-      <frameworkAssembly assemblyName="mscorlib" targetFramework=".NETFramework4.6.1" />
-      <frameworkAssembly assemblyName="System" targetFramework=".NETFramework4.6.1" />
-      <frameworkAssembly assemblyName="System.Core" targetFramework=".NETFramework4.6.1" />
-    </frameworkAssemblies>
   </metadata>
   <files>
-    <file src="C:\Users\anagniho\.nuget\packages\microsoft.dotnet.arcade.sdk\6.0.0-beta.21105.12\tools\Assets\DotNetPackageIcon.png" target="Icon.png" exclude="" />
-    <file src="C:\git\runtime5\artifacts\bin\Microsoft.Extensions.Configuration.Binder\net461-Debug\Microsoft.Extensions.Configuration.Binder.dll" target="lib\net461" exclude="" />
-    <file src="C:\git\runtime5\artifacts\bin\Microsoft.Extensions.Configuration.Binder\net461-Debug\Microsoft.Extensions.Configuration.Binder.pdb" target="lib\net461" exclude="" />
-    <file src="C:\Users\anagniho\.nuget\packages\microsoft.private.intellisense\5.0.0-preview-20201009.2\IntellisenseFiles\net\1033\Microsoft.Extensions.Configuration.Binder.xml" target="lib\net461" exclude="" />
-    <file src="C:\git\runtime5\artifacts\bin\Microsoft.Extensions.Configuration.Binder\netstandard2.0-Debug\Microsoft.Extensions.Configuration.Binder.dll" target="lib\netstandard2.0" exclude="" />
-    <file src="C:\git\runtime5\artifacts\bin\Microsoft.Extensions.Configuration.Binder\netstandard2.0-Debug\Microsoft.Extensions.Configuration.Binder.pdb" target="lib\netstandard2.0" exclude="" />
-    <file src="C:\Users\anagniho\.nuget\packages\microsoft.private.intellisense\5.0.0-preview-20201009.2\IntellisenseFiles\net\1033\Microsoft.Extensions.Configuration.Binder.xml" target="lib\netstandard2.0" exclude="" />
-    <file src="C:\git\runtime5\LICENSE.TXT" target="LICENSE.TXT" exclude="" />
-    <file src="C:\git\runtime5\THIRD-PARTY-NOTICES.TXT" target="THIRD-PARTY-NOTICES.TXT" exclude="" />
+    <file src="C:\git\runtime4\artifacts\bin\Microsoft.Extensions.Configuration.Binder\net461-Debug\Microsoft.Extensions.Configuration.Binder.dll" target="lib\net461\Microsoft.Extensions.Configuration.Binder.dll" />
+    <file src="C:\git\runtime4\artifacts\bin\Microsoft.Extensions.Configuration.Binder\net461-Debug\Microsoft.Extensions.Configuration.Binder.xml" target="lib\net461\Microsoft.Extensions.Configuration.Binder.xml" />
+    <file src="C:\git\runtime4\artifacts\bin\Microsoft.Extensions.Configuration.Binder\netstandard2.0-Debug\Microsoft.Extensions.Configuration.Binder.dll" target="lib\netstandard2.0\Microsoft.Extensions.Configuration.Binder.dll" />
+    <file src="C:\git\runtime4\artifacts\bin\Microsoft.Extensions.Configuration.Binder\netstandard2.0-Debug\Microsoft.Extensions.Configuration.Binder.xml" target="lib\netstandard2.0\Microsoft.Extensions.Configuration.Binder.xml" />
+    <file src="C:\Users\anagniho\.nuget\packages\microsoft.dotnet.arcade.sdk\6.0.0-beta.21105.12\tools\Assets\DotNetPackageIcon.png" target="Icon.png" />
+    <file src="C:\git\runtime4\LICENSE.TXT" target="LICENSE.TXT" />
+    <file src="C:\git\runtime4\THIRD-PARTY-NOTICES.TXT" target="THIRD-PARTY-NOTICES.TXT" />
   </files>
 </package>
\ No newline at end of file
```

File: Directory.Build.props (hunk 3 above)
Comment (ViktorHofer) [reply Directory.Build.props:230]: This is interesting. I don't have context on why Arcade defaults that to true. Disregard then.

File: Directory.Build.props (hunk 6 above)
Comment (ViktorHofer) [reply Directory.Build.props:226]: Also, Arcade already provides the package icon path which I guess will be used to embed the icon: https://github.com/dotnet/arcade/blob/99b9f805da070d301d76eac578a46dd75d9e5f7c/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectDefaults.props#L21.

File: src/libraries/Microsoft.Extensions.Configuration.Xml/src/Microsoft.Extensions.Configuration.Xml.csproj [hunk 8]
```diff
@@ -3,14 +3,15 @@
...
+    <IsPackable>true</IsPackable>
   </PropertyGroup>
 
   <ItemGroup>
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Configuration\src\Microsoft.Extensions.Configuration.csproj" />
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Configuration.Abstractions\src\Microsoft.Extensions.Configuration.Abstractions.csproj" />
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Configuration.FileExtensions\src\Microsoft.Extensions.Configuration.FileExtensions.csproj" />
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.FileProviders.Abstractions\src\Microsoft.Extensions.FileProviders.Abstractions.csproj" />
-    <ProjectReference Include="$(LibrariesProjectRoot)System.Security.Cryptography.Xml\src\System.Security.Cryptography.Xml.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Security.Cryptography.Xml\src\System.Security.Cryptography.Xml.csproj" Condition="!$(TargetFramework.StartsWith('net4'))" />
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.Extensions.Configuration.Xml/src/Microsoft.Extensions.Configuration.Xml.csproj:14]:
```suggestion
    <ProjectReference Include="$(LibrariesProjectRoot)System.Security.Cryptography.Xml\src\System.Security.Cryptography.Xml.csproj" Condition="'$(TargetFramework)' == 'netstandard2.0'" />
```

File: src/libraries/System.ServiceProcess.ServiceController/ref/System.ServiceProcess.ServiceController.csproj [hunk 9]
```diff
@@ -14,6 +14,6 @@
     <Reference Include="System.ServiceProcess" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\..\System.Diagnostics.EventLog\ref\System.Diagnostics.EventLog.csproj" />
+    <ProjectReference Include="..\..\System.Diagnostics.EventLog\ref\System.Diagnostics.EventLog.csproj" Condition="!$(TargetFramework.StartsWith('net4'))" />
```
Comment (ViktorHofer) [inline src/libraries/System.ServiceProcess.ServiceController/ref/System.ServiceProcess.ServiceController.csproj:17]:
```suggestion
    <ProjectReference Include="..\..\System.Diagnostics.EventLog\ref\System.Diagnostics.EventLog.csproj" Condition="'$(TargetFramework)' == 'netstandard2.0'" />
```

File: src/libraries/libraries-packages.proj [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.Build.Traversal" DefaultTargets="Pack">
 
   <PropertyGroup>
-    <TraversalGlobalProperties>BuildAllProjects=true</TraversalGlobalProperties>
+    <TraversalGlobalProperties>BuildAllProjects=true;BuildTargetFramework=</TraversalGlobalProperties>
```
Comment (ViktorHofer) [inline src/libraries/libraries-packages.proj:4]: why is this change necessary?

**Comment by ViktorHofer** [comment]
Curious, why is `<Serviceable>true</Serviceable>` removed in the nuspec? Arcade sets it to true: https://github.com/dotnet/arcade/blob/99b9f805da070d301d76eac578a46dd75d9e5f7c/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectDefaults.props#L14. Is it omitted in case it's true?

File: src/libraries/Directory.Build.targets (hunk 5 above)
Comment (ViktorHofer) [reply src/libraries/Directory.Build.targets:237]: I now better understand why it's uncertain that we want to use Arcade's default version calculation. Currently we hardcode the PackageVersion to $(MajorVersion).$(MinorVersion).0 and manually rev the PackageVersion for projects that are part of an incremental servicing release.

That said I tend towards let Arcade default the `PackageVersion` to `Version` which for the next servicing releases as an example is 5.0.5. Packages that incrementally service specify a `PackageVersion` anyway and if they don't they default to the current patch digit. I see advantages of that mechanism over the hardcoded `.0` patch version which would actually cause publishing to fail if no `PackageVersion` is specified (only talking about NuGet pack task without pkgproj and packageIndex infra).

Any concerns about that? cc @ericstj

Comment (ViktorHofer) [reply src/libraries/Directory.Build.targets:237]: Talked with @ericstj about this offline and we agree that defaulting the PackageVersion to Version (SDK default behavior) is the right approach. This will mean that by default, i.e. for 5.0.5 servicing, the PackageVersion will be 5.0.5. As we incrementally service packages anyway, the `PatchVersion` will need to be set explicitly anyway (likely via an inferred property to also handle the required AssemblyVersion bump).

File: Directory.Build.props (hunk 2 above)
Comment (ViktorHofer) [reply Directory.Build.props:233]: @ericstj thoughts? I would just let NuGet default to PackageId and only specify a title for packages that explicitly want to.

File: eng/packaging.props [hunk 11]
```diff
@@ -15,6 +15,10 @@
     <SkipGenerationCheck>true</SkipGenerationCheck>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <NuSpecOutputPath Condition="'$(NuSpecOutputPath)' == ''">$([MSBuild]::NormalizeDirectory('$(ArtifactsPackagesDir)', 'specs'))</NuSpecOutputPath>
```
Comment (ViktorHofer) [inline eng/packaging.props:19]: Are nuspecs generated by default by NuGet when invoking their Pack task?

File: Directory.Build.props (hunk 2 above)
Comment (ericstj) [reply Directory.Build.props:233]: We should avoid customizations unless they are absolutely necessary.  If we are somehow breaking the default behavior we should fix that rather than duplicate the defaults.

Comment (ViktorHofer) [reply Directory.Build.props:233]: image

Right. So my suggestion is to remove L233.

File: src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Microsoft.Extensions.Caching.Abstractions.csproj (hunk 7 above)
Comment (ViktorHofer) [reply src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Microsoft.Extensions.Caching.Abstractions.csproj:6]: IMO I would prefer this property to be explicit

File: src/libraries/Directory.Build.targets (hunk 5 above)
Comment (ericstj) [reply src/libraries/Directory.Build.targets:237]: I believe the conclusion was that  in servicing all libraries need to by default be frozen to Major.0.0 since we will not be shipping them by default.  This ensures that P2P references get the correct value for CSProjs that are being packed.  Then as we decide to ship a library it can declare the new servicing version as a single integer and we can use that to compute both the PackageVersion and AssemblyVersion (as appropriate).

Comment (ViktorHofer) [reply src/libraries/Directory.Build.targets:237]: Oh gosh, I forgot about that part of the discussion, you are absolutely right.

I propose to remove the line above and only hardcode the `PackageVersion` (in Versions.props) during servicing to $(MajorVersion).$(MinorVersion).0. We should make sure that properties like `PackageVersion` that apply to every package in the repository are defined centrally (ie in Directory.Build.* files or Versions.props). Related: https://github.com/dotnet/runtime/issues/48229

File: eng/packaging.props (hunk 11 above)
Comment (Anipik) [reply eng/packaging.props:19]: Yes, by default its in the obj directory.

**Comment by ericstj** [comment]
What's the status of this PR?  It would have been nice to have as it would save some complexity in https://github.com/dotnet/runtime/pull/51064.

**Comment by Anipik** [comment]
Its currently blocked on this https://github.com/NuGet/NuGet.Client/pull/3980 i need to fix the tests in the nuget pr

**Comment by ericstj** [comment]
> Its currently blocked on this NuGet/NuGet.Client#3980 i need to fix the tests in the nuget pr

Do we have an issue on our side somewhere describing why we need that or how we intend to use it?  Trying to connect the dots here.

**Comment by ViktorHofer** [comment]
@Anipik can't you just in the meantime override the property via a target? There are places where we already do stuff in a target fo packaging related settings/validations.

**Comment by Anipik** [comment]
closing this in favor of https://github.com/dotnet/runtime/pull/51765
